receiver/k8slogreceiver/config.go

```diff
@@ -0,0 +1,217 @@
+	// For example, if your pod spec contains the following labels,
+	//
+	// app.kubernetes.io/component: mysql
+	// app.kubernetes.io/version: 5.7.21
+	//
+	// and you'd like to add tags for all labels with prefix app.kubernetes.io/ and also trim the prefix,
+	// then you can specify the following extraction rules:
+	//
+	// extract:
+	//   labels:
+	//     - tag_name: $$1
+	//       key_regex: kubernetes.io/(.*)
+	//
+	// this will add the `component` and `version` tags to the spans or metrics.
+	TagName string `mapstructure:"tag_name"`
+
+	// Key represents the key (annotation, label or etc.) name. It uses exact match.
+	Key string `mapstructure:"key"`
+	// KeyRegex is a regular expression used to extract a Key that matches the regex.
+	// Out of Key or KeyRegex, only one option is expected to be configured at a time.
+	KeyRegex string `mapstructure:"key_regex"`
+
+	// Regex is an optional field used to extract a sub-string from a complex field value.
+	// The supplied regular expression must contain one named parameter with the string "value"
+	// as the name. For example, if your pod spec contains the following annotation,
+	//
+	// kubernetes.io/change-cause: 2019-08-28T18:34:33Z APP_NAME=my-app GIT_SHA=58a1e39 CI_BUILD=4120
+	//
+	// and you'd like to extract the GIT_SHA and the CI_BUILD values as tags, then you must
+	// specify the following two extraction rules:
+	//
+	// extract:
+	//   annotations:
+	//     - tag_name: git.sha
+	//       key: kubernetes.io/change-cause
+	//       regex: GIT_SHA=(?P<value>\w+)
+	//     - tag_name: ci.build
+	//       key: kubernetes.io/change-cause
+	//       regex: JENKINS=(?P<value>[\w]+)
+	//
+	// this will add the `git.sha` and `ci.build` resource attributes.
+	Regex string `mapstructure:"regex"`
+
+	// From represents the source of the labels/annotations.
+	// Env is always from container environment variables.
+	// Supported values:
+	// - pod (default): extract from pod labels/annotations.
+	// May be supported in the future:
+	// - namespace
+	From string `mapstructure:"from"`
+}
+
+func (c Config) Validate() error {
+	return c.Discovery.Validate()
+}
+
+// SourceConfig allows specifying how to discover containers to collect logs from.
+type SourceConfig struct {
+	// Mode represents the mode of the k8slog receiver.
+	// Valid values are:
+	// - "daemonset-stdout": (default) otel is deployed as a daemonset and collects logs from stdout of containers.
+	//
+	// Will be supported in the future:
+	// - "daemonset-file": otel is deployed as a daemonset and collects logs from files inside containers.
+	// - "sidecar": otel is deployed as a sidecar and collects logs from files.
+	Mode string `mapstructure:"mode"`
+
+	// NodeFromEnv represents the environment variable which contains the node name.
+	NodeFromEnv string `mapstructure:"node_from_env"`
+
+	// HostRoot represents the path which is used to mount the host's root filesystem.
```

Comment:
Yes. But the files under `/var/log/pods` may be symbol links to paths defined by runtimes, which can also be configured by ops. If they don't mount the whole host's root directory, they must have knowledge about will the files actually live. This may be difficult for users, especially with k8s clusters that mix nodes with different runtimes.

Reply:
Wouldn't that be better if we chose a reasonable list of directories like `/var/log/pods/*`, `/var/lib/docker/*` etc  in order to specifically define the possible directories the known runtimes use? 
We can even make this list configurable so as advanced users could set their own extra directories if the want.

Mounting the whole host's root directory looks problematic to me and maybe shouldn't be the default.
In GKE Autopilot for example this would not be supported: https://github.com/grafana/loki/issues/9100.

So my suggestion would be to start with reasonable defaults, while also giving the option for more advanced users to extend these defaults.